data/transactions/logic/assembler.go

```diff
@@ -248,8 +249,50 @@ type OpStream struct {
 	HasStatefulOps bool
 }
 
-// createLabel inserts a label reference to point to the next
-// instruction, reporting an error for a duplicate.
+// ProgramKnowledge tracks statically known information as we assemble
+type ProgramKnowledge struct {
+	// known is the stack of known stack types, based on specs of opcodes seen.
```

Comment:
@jannotti Minor:  There's a few inline comment references to _known_ that I'm not following.  I think the comment intends to reference `stack`.  Though it's unclear to me if the intent is rename `stack` to `known` or vice-versa?

Reply:
My mistake.  The `stack` used to be called `known`. I'll clean it up.